pkg/osquery/tables/firefox_preferences/table.go

```diff
@@ -0,0 +1,116 @@
+package firefox_preferences
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/launcher/pkg/dataflatten"
+	"github.com/kolide/launcher/pkg/osquery/tables/dataflattentable"
+	"github.com/kolide/launcher/pkg/osquery/tables/tablehelpers"
+	"github.com/osquery/osquery-go/plugin/table"
+)
+
+type Table struct {
+	name   string
+	logger log.Logger
+}
+
+const tableName = "kolide_firefox_preferences"
+
+var re = regexp.MustCompile(`^user_pref\("([^,]+)",\s*"?(.*?)"?\);$`)
+
+func TablePlugin(logger log.Logger) *table.Plugin {
+	columns := dataflattentable.Columns(
+		table.TextColumn("path"),
+	)
+
+	t := &Table{
+		name:   tableName,
+		logger: logger,
+	}
+
+	return table.NewPlugin(t.name, columns, t.generate)
+}
+
+func (t *Table) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	return generateData(queryContext, t.logger), nil
```

Comment:
Why have both `generate` and `generateData` ?

Reply:
I left it like this for testing purposes, since `generate` is hung off the `Table` struct. Talked with @James-Pickett about this while he was helping me with tests, but if you have a suggestion on how to work around that, I'm happy to change it.

Reply:
I felt like having the function with the full signature of `func (t *Table) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error)` would just add noise to the tests and the func itself. What we are really doing / testing here is parsing prefs.js

Reply:
Not sure what you mean James...

Usually I'd just instantiate the table `table := Table{logger: log.NewNopLogger()}` and then call `table.generate(...)`.  It's not really different than calling `generateData` here.

Reply:
I'll expand a bit .... `generate()` requires the `ctx context.Context` parameter and the `error` return value. Both of these are irrelevant to the actual parsing of prefs.js in `generateData()` since we don't use context or ever return an error there. I feel like it just makes it slightly more difficult for a human to parse. But this point is not a show stopper for me if you feel strongly about it.

Reply:
Yeah... You're not wrong. I dunno. I guess my bias it to have fewer functions. But if this was larger I would 100% be with you about splitting it into testable pieces